conversations.go

```diff
@@ -0,0 +1,175 @@
+package main
+
+import (
+	"accounts-service/auth"
+	"accounts-service/models"
+	conversationsv1 "accounts-service/protorepo/noted/accounts/v1"
+	"accounts-service/validators"
+	"context"
+
+	"go.uber.org/zap"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type conversationsAPI struct {
+	conversationsv1.UnimplementedConversationsAPIServer
+
+	auth   auth.Service
+	logger *zap.Logger
+	repo   models.ConversationsRepository
+}
+
+var _ conversationsv1.ConversationsAPIServer = &conversationsAPI{}
+
+func (server *conversationsAPI) CreateConversation(ctx context.Context, in *conversationsv1.CreateConversationRequest) (*conversationsv1.CreateConversationResponse, error) {
+	err := validators.ValidateCreateConversationRequest(in)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	groupConversations, err := server.repo.List(ctx, &models.AllConversationsFilter{ID: in.GroupId})
+
+	for _, conv := range groupConversations {
+		res := conv.Title == in.Title
+		if res {
+			return nil, status.Error(codes.Internal, "a conversations with this title always exist")
+		}
+	}
+
+	conversation, err := server.repo.Create(ctx, &models.ConversationInfo{Title: in.Title, GroupID: in.GroupId})
+	if err != nil {
+		return nil, statusFromModelError(err)
+	}
+	return &conversationsv1.CreateConversationResponse{
+		Conversation: &conversationsv1.Conversation{
+			Id:      conversation.ID,
+			GroupId: conversation.GroupID,
+			Title:   conversation.Title,
+		},
+	}, nil
+}
+
+func (server *conversationsAPI) GetConversation(ctx context.Context, in *conversationsv1.GetConversationRequest) (*conversationsv1.GetConversationResponse, error) {
+	err := validators.ValidateGetConversationRequest(in)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, "could not get Conversation")
+	}
```

Comment:
You forgot to authenticate the user. Only a user which is part of the group can access a given conversation. You should add this authorization check to the endpoint code.